src/main/java/net/fabricmc/loom/MixinAnnotationProcessorExtension.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * This file is part of fabric-loom, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom;
+
+import java.io.File;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.gradle.api.Action;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.UnknownTaskException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.plugins.ExtraPropertiesExtension;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.util.PatternSet;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A gradle extension to configure mixin annotation processor.
+ */
+public class MixinAnnotationProcessorExtension {
+	public static final String MIXIN_INFORMATION_CONTAINER = "mixin";
+	private boolean isDefault;
+
+	private final Project project;
+
+	/**
+	 * A information container stores necessary information
+	 * for configure mixin annotation processor. It stores
+	 * in [SourceSet].ext.mixin.
+	 */
+	public static final class MixinInformationContainer {
+		private final SourceSet sourceSet;
+		private final String refmapName;
+		private final PatternSet mixinJsonPattern;
+
+		private Stream<String> mixinJsonNames;
```

Comment:
Why a Stream and not something more conventional like a set or list?

Reply:
Because it was from `Stream` (see code below)
```Java
Stream<String> mixinJsonNames = sourceSet.getResources().matching(pattern).getFiles().stream().map(File::getName);
```
and will be used as `Stream`
```Java
mixinJsonNames.map(
```
There is unnecessary to collect it then stream it.